src/engine/enginebuffer.cpp

```diff
@@ -1408,6 +1408,7 @@ void EngineBuffer::notifyTrackLoaded(
     // Note: we are still in a worker thread.
     for (const auto& pControl: qAsConst(m_engineControls)) {
         pControl->trackLoaded(pNewTrack);
+        pControl->setCurrentSample(0.0, m_pTrackSamples->get(), m_pTrackSampleRate->get());
```

Comment:
m_filepos_play instead of 0.0? This would be consistent with the other invocation.

Reply:
This makes the tests fail, probably because `m_filepos_play` is set to `DBL_MIN` for some reason...
https://github.com/Holzhaus/mixxx/actions/runs/597008525

Reply:
That's the is the value it is set to when loading the track. Using 0.0 is inconsistent, although it works.

All this code is really hacky. I don't care, we are not able to fix those issues. Use 0.0 if it works then let's use it. Maybe leave a comment that m_filepos_play does not work for unknown reasons.

Reply:
But isn't 4.2045991279448952e-313 almost zero? EXPECT_DOUBLE_EQ might not be appropriate for the test, however.

Reply:
The value of dCurrentSample should relay don't matter here. Before this PR it has the last seek value form the previous loaded track or 0.0 for this is the first track. If setting this to any other value here fails a test, the test is broken, because it probably relies on the initial value of 0.0 without any previous loaded track. 

In this regards it is already inconsistent that m_filepos_play and dCurrentSample are initialized differently. 

I will have a look how we can fix it.

Reply:
In general, we should set the values to be consistent in the code.  If that makes the tests fail, the tests should be updated.  We should never set values to match tests when that conflicts with intuition or consistency

Reply:
Unfortunately, it is not obvious when a state is considered "consistent". Too many side-effects and implicit assumptions.